frontend/scripts/tests/puppeteer/profile-actor.test.js

```diff
@@ -0,0 +1,274 @@
+/* eslint-disable no-console */
+import puppeteer from 'puppeteer';
+
+import mocks from '../mocks';
+
+let page;
+let browser;
+const baseUrl = 'http://0.0.0.0:8081';
+
+const openBrowser = visible =>
+  visible
+    ? {
+        headless: false,
+        slowMo: 80,
+        args: [`--window-size=1920,1080`]
+      }
+    : { args: ['--no-sandbox'] };
+
+beforeAll(async () => {
+  browser = await puppeteer.launch(openBrowser(false));
+  page = await browser.newPage();
+  await page.setRequestInterception(true);
+  page.on('request', interceptedRequest => {
+    const url = interceptedRequest
+      .url()
+      .replace('https:', '')
+      .replace('http:', '');
+
+    if (url in mocks) {
+      console.warn(`URL (${url}) intercepted and response mocked`);
+      setTimeout(
+        () =>
+          interceptedRequest.respond({
+            status: 200,
+            contentType: 'application/json',
+            headers: {
+              'Access-Control-Allow-Origin': '*'
+            },
+            body: JSON.stringify(mocks[url])
+          }),
+        300
+      );
+    } else {
+      interceptedRequest.continue();
+    }
+  });
+});
+
+afterAll(() => {
+  browser.close();
+});
+
+describe('Profile Root search', () => {
+  test(
+    'search for bunge and click importer result',
+    async () => {
+      const nodeName = 'bunge';
+      const nodeType = 'importer';
+      const profileType = 'actor';
+
+      await page.goto(`${baseUrl}/profiles`);
+      await page.waitForSelector('[data-test=profile-search]');
+      await page.click('[data-test=search-input-desktop]');
+      await page.type('[data-test=search-input-desktop]', nodeName);
+      await page.waitForSelector(`[data-test=search-result-${nodeType}-${nodeName}]`);
+      await page.click(`[data-test=search-result-${nodeType}-${nodeName}]`);
+
+      expect(page.url().startsWith(`${baseUrl}/profile-${profileType}`)).toBe(true);
+
+      expect.assertions(1);
+    },
+    30000
```

Comment:
what's the reason behind it? we shouldn't rely on wait times

Reply:
So 30 seconds it's the puppeteer timeout limit, and the tests take some time to be evaluated, more that the `jest.timeout` default. So I juust bumped it to the puppeteer timeout, so that if the test fails is because the browser timed-out and not jest.

Reply:
in that case maybe you can create `cleanTest` function that wraps up test function and adds the timeout limit and use that one instead of `test`

Reply:
and `cleanTest` could land in the helpers folder! :D